index.js

```diff
@@ -108,8 +112,13 @@ class MultiStream extends stream.Readable {
     }
 
     const onClose = () => {
-      if (!stream._readableState.ended) {
-        this.destroy()
+      const readableEnded =
+        (stream._readableState && stream._readableState.ended) ||
+        stream.readableEnded
```

Comment:
_readableState is always set as all streams are run through the toStreams2 thing above. i don't see why the readableEnded check is needed but can't hurt i guess :)